README.md

```diff
@@ -227,11 +227,21 @@ docker run \
 > * The Pravega storage doesn't support reading the stream events in the random order.
 > * Works synchronously
 
-There is also another option, called `storage-driver-read-timeoutMillis`. Pravega documentation says it only works when
+There is a specific option (config parameter) for reading called `storage-driver-event-timeoutMillis`. Pravega documentation says it only works when
 there is no available event in the stream. `readNextEvent()` will block for the specified time in ms. So, in theory 0
 and 1 should work just fine. They do not. In practice, this value should be somewhere between 100 and 2000 ms (2000 is
 Pravega default value).
 
+As Pravega is a streaming storage, it does not provide any information on a stream size. 
+So, we allocate memory for reading one event only and then repeat it until we meet the end of a stream.
+In terms of Mongoose it is called recycle-mode and activated via `--load-op-recycle` option.
+That makes this option obligatory for doing read operations on events.
+Learn more about recycle-mode here: https://github.com/emc-mongoose/mongoose-base/tree/master/doc/design/recycle_mode.
+
+There is one important fact considering the end of a stream. As we do not know the size of a stream,
+we only find out that it is the end, when `readNextEvent()` returns false, so 1 failed operation is not
```

Comment:
`readNextEvent` returns the `EventRead` instance which returns `null` on `getEvent` call